client/src/features/groupsV2/new/GroupNew.tsx

```diff
@@ -0,0 +1,286 @@
+  );
+  const dispatch = useDispatch();
+  const toggleModal = useCallback(() => {
+    dispatch(toggleGroupCreationModal());
+  }, [dispatch]);
+
+  return (
+    <>
+      <Modal
+        backdrop="static"
+        centered
+        data-cy="new-group-modal"
+        fullscreen="lg"
+        isOpen={showGroupCreationModal}
+        scrollable
+        size="lg"
+        unmountOnClose={true}
+        toggle={toggleModal}
+      >
+        <ModalHeader
+          data-cy="new-group-modal-header"
+          tag="div"
+          toggle={toggleModal}
+        >
+          <h2>Create a new group</h2>
+          <p className={cx("fs-6", "mb-0")}>
+            Groups let you group together related projects and control who can
+            access them.
+          </p>
+        </ModalHeader>
+
+        <div data-cy="create-new-group-content">
+          {user.logged ? (
+            <GroupV2CreationDetails />
+          ) : (
+            <ModalBody>
+              <LoginAlert
+                logged={user.logged}
+                textIntro="Only authenticated users can create new groups."
+                textPost="to create a new group."
+              />
+            </ModalBody>
+          )}
+        </div>
+      </Modal>
+    </>
+  );
+}
+
+function GroupV2CreationDetails() {
+  const [isCollapseOpen, setIsCollapseOpen] = useState(false);
+  const toggleCollapse = () => setIsCollapseOpen(!isCollapseOpen);
+
+  const [createGroup, result] = usePostGroupsMutation();
+  const navigate = useNavigate();
+
+  const dispatch = useDispatch();
+  const toggleModal = useCallback(() => {
+    dispatch(toggleGroupCreationModal());
+  }, [dispatch]);
+
+  // Form initialization
+  const {
+    control,
+    formState: { dirtyFields, errors },
+    handleSubmit,
+    setValue,
+    watch,
+  } = useForm<GroupPostRequest>({
+    mode: "onChange",
+    defaultValues: {
+      description: "",
+      name: "",
+      slug: "",
+    },
+  });
+
+  // We watch for changes in the name and derive the slug from it
+  const currentName = watch("name");
+  useEffect(() => {
+    setValue("slug", slugFromTitle(currentName, true, true), {
+      shouldValidate: true,
+    });
+  }, [currentName, setValue]);
+
+  // Slug is use to show the projected URL
+  const currentSlug = watch("slug");
+
+  // Group creation utilities
+  const onSubmit = useCallback(
+    (groupPostRequest: GroupPostRequest) => {
+      createGroup({ groupPostRequest });
+    },
+    [createGroup]
+  );
+
+  useEffect(() => {
+    if (result.isSuccess) {
+      const groupUrl = generatePath(ABSOLUTE_ROUTES.v2.groups.show.root, {
+        slug: currentSlug,
+      });
+      navigate(groupUrl);
+      dispatch(setGroupCreationModal(false));
+    }
+  }, [currentSlug, result, dispatch, navigate]);
+
+  const nameHelpText = (
+    <FormText className="input-hint">
+      The URL for this group will be{" "}
+      <span className="fw-bold">
+        renkulab.io/v2/groups/{currentSlug || "<Name>"}
```

Comment:
We lowercase the name:

```suggestion
        renkulab.io/v2/groups/{currentSlug || "<name>"}
```

Reply:
Not sure I get this comment. That's only for the user's convenience to show a placeholder while the name isn't filled in. They see the upper case `Name` on the label, I don't think making it lowercase improves the UX

![image](https://github.com/user-attachments/assets/c8dd806c-ddc6-45f9-88f9-b5296b547dd6)

Reply:
Done